File: app/src/main/java/social/entourage/android/user/UserOrganizationsAdapter.java
package social.entourage.android.user;

import android.net.Uri;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.List;

import social.entourage.android.R;
import social.entourage.android.api.model.BaseOrganization;
import social.entourage.android.api.model.Partner;
import social.entourage.android.api.tape.Events;
import social.entourage.android.tools.BusProvider;

/**
 * Created by mihaiionescu on 24/03/16.
 */
public class UserOrganizationsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    public static class OrganizationViewHolder extends RecyclerView.ViewHolder {

        public TextView mOrganizationName;
        public ImageView mOrganizationLogo;
        public TextView mOrganizationType;
        public View mSeparator;

        public Partner partner;

        public OrganizationViewHolder(View v) {
            super(v);
            mOrganizationName = v.findViewById(R.id.organization_name);
            mOrganizationType = v.findViewById(R.id.organization_type);
            mOrganizationLogo = v.findViewById(R.id.organization_logo);
            mSeparator = v.findViewById(R.id.organization_separator);

            if (mOrganizationLogo != null) {
                mOrganizationLogo.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(final View view) {
                        if (partner != null) {
                            BusProvider.INSTANCE.getInstance().post(new Events.OnPartnerViewRequestedEvent(partner));
                        }
                    }
                });
            }
        }

    }

    private List<BaseOrganization> organizationList;

    public UserOrganizationsAdapter(List<BaseOrganization> organizationList) {
        this.organizationList = organizationList;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_user_profile_organization, parent, false);
        return new OrganizationViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull final RecyclerView.ViewHolder holder, final int position) {
        BaseOrganization organization = organizationList.get(position);
        OrganizationViewHolder organizationViewHolder = (OrganizationViewHolder) holder;

        organizationViewHolder.mOrganizationName.setText(organization.getName());

        organizationViewHolder.mOrganizationType.setText(organization.getTypeAsString(organizationViewHolder.mOrganizationType.getContext()));

        String organizationLogo = organization.getLargeLogoUrl();
        if (organizationLogo != null) {
            Picasso.get()
                    .load(Uri.parse(organizationLogo))
                    .into(organizationViewHolder.mOrganizationLogo);
        }
        else {
            organizationViewHolder.mOrganizationLogo.setImageDrawable(null);
        }

        if (organization.getType() == BaseOrganization.TYPE_PARTNER) {
            organizationViewHolder.partner = (Partner)organization;
        }
        else {
            organizationViewHolder.partner = null;
        }

        organizationViewHolder.mSeparator.setVisibility(position == organizationList.size() - 1 ? View.GONE : View.VISIBLE);
    }

    @Override
    public int getItemCount() {
        if (organizationList == null) return 0;
        return organizationList.size();
    }

    public void setOrganizationList(final List<BaseOrganization> organizationList) {
        this.organizationList = organizationList;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/social/entourage/android/user/UserOrganizationsAdapter.kt
package social.entourage.android.user

import android.net.Uri
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.layout_user_profile_organization.view.*
import social.entourage.android.R
import social.entourage.android.api.model.BaseOrganization
import social.entourage.android.api.model.Partner
import social.entourage.android.api.tape.Events.OnPartnerViewRequestedEvent
import social.entourage.android.tools.BusProvider

/**
 * Created by mihaiionescu on 24/03/16.
 */
class UserOrganizationsAdapter(private var organizationList: List<BaseOrganization>?) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    class OrganizationViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var partner: Partner? = null

        init {
            itemView.ui_layout_click?.setOnClickListener {
                partner?.let {
                    BusProvider.instance.post(OnPartnerViewRequestedEvent(it))
                }
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.layout_user_profile_organization, parent, false)
        return OrganizationViewHolder(v)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        organizationList?.get(position)?.let { organization ->
            val organizationViewHolder = holder as OrganizationViewHolder
            organizationViewHolder.itemView.organization_name.text = organization.name
            organizationViewHolder.itemView.organization_type.text = organization.getTypeAsString(organizationViewHolder.itemView.context)
            organization.largeLogoUrl?.let {organizationLogo ->
                Picasso.get()
                        .load(Uri.parse(organizationLogo))
                        .into(organizationViewHolder.itemView.organization_logo!!)
            } ?: run {
                organizationViewHolder.itemView.organization_logo?.setImageDrawable(null)
            }
            organizationViewHolder.partner = (organization as? Partner)
            organizationViewHolder.itemView.organization_separator.visibility = if (position == organizationList!!.size - 1) View.GONE else View.VISIBLE
        }
    }

    override fun getItemCount(): Int {
        return organizationList?.size ?: 0
    }

    fun setOrganizationList(organizationList: List<BaseOrganization>?) {
        this.organizationList = organizationList
        notifyDataSetChanged()
    }

}